src/field/field3d.cxx

```diff
@@ -514,17 +514,12 @@ void Field3D::applyParallelBoundary() {
   TRACE("Field3D::applyParallelBoundary()");
 
   checkData(*this);
+  ASSERT1(!yup_fields.empty());
```

Comment:
Is it worth checking `ydown_fields` too? They should always be initialised together, but you never know...

Reply:
`hasParallelSlices()` does this, that might also be more explicit:

```suggestion
  ASSERT1(hasParallelSlices());
```